### ISSUE TITLE
Issue with AmpersandsPipesVsAndOr?

### ISSUE THREAD

**Issue by lee-dohm** [rubocop/rubocop#185]
I have the following code in my current project that is triggering this rule (among other locations):

``` ruby
def ==(other)
  return false unless other.respond_to?(:from) && other.respond_to?(:to) && other.respond_to?(:style)

  @from == other.from && @to == other.to && @style == other.style
end
```

Specifically, the last line is being called out as the culprit. I'm not using the Boolean operators for flow control here, I am constructing a Boolean expression and returning the result. 

Can we reliably tell the difference between the Boolean expression `foo || bar || baz` and the Boolean expression but should be control flow statement `foo || bar || baz`?
